src/server/scripts/Northrend/Naxxramas/boss_razuvious.cpp

```diff
@@ -70,36 +67,61 @@ class boss_razuvious : public CreatureScript
     {
         boss_razuviousAI(Creature* creature) : BossAI(creature, BOSS_RAZUVIOUS) { }
 
-        void KilledUnit(Unit* /*victim*/) override
+        void SummonAdds()
         {
-            if (!(rand32() % 3))
-                DoPlaySoundToSet(me, SOUND_SLAY);
+            me->SummonCreatureGroup(SUMMON_GROUP_10MAN);
+            if (Is25ManRaid())
+                me->SummonCreatureGroup(SUMMON_GROUP_25MAN);
         }
 
-        void DamageTaken(Unit* pDone_by, uint32& uiDamage) override
+        void InitializeAI() override
         {
-            // Damage done by the controlled Death Knight understudies should also count toward damage done by players
-            if (pDone_by->GetTypeId() == TYPEID_UNIT && (pDone_by->GetEntry() == 16803 || pDone_by->GetEntry() == 29941))
+            if (!me->isDead())
             {
-                me->LowerPlayerDamageReq(uiDamage);
+                Reset();
+                SummonAdds();
             }
         }
 
+        void JustReachedHome() override
+        {
+            _JustReachedHome();
+            SummonAdds();
+            me->GetMotionMaster()->Initialize();
+        }
+
+        void KilledUnit(Unit* victim) override
+        {
+            if (victim->GetTypeId() == TYPEID_PLAYER || (victim->GetTypeId() == TYPEID_UNIT && victim->GetEntry() == NPC_DK_UNDERSTUDY))
+                Talk(SAY_SLAY);
+        }
+
+        void SpellHit(Unit* caster, SpellInfo const* spell) override
+        {
+            if (spell->Id == SPELL_UNDERSTUDY_TAUNT)
+                Talk(SAY_TAUNTED, caster);
+        }
+
         void JustDied(Unit* /*killer*/) override
         {
-            _JustDied();
-            DoPlaySoundToSet(me, SOUND_DEATH);
-            me->CastSpell(me, SPELL_HOPELESS, true); /// @todo this may affect other creatures
+            Talk(SAY_DEATH);
+            me->CastSpell(me, SPELL_HOPELESS, true);
+
+            events.Reset();
+            scheduler.CancelAll();
```

Comment:
also, why that line? you are not using TaskScheduler anywhere in script, so not needed to reset nothing